File: CinemaNow/CinemaNow.Services/Database/Reservation.cs
using System;
using System.Collections.Generic;

namespace CinemaNow.Services.Database;

public partial class Reservation
{
    public int Id { get; set; }

    public int? UserId { get; set; }

    public int? ScreeningId { get; set; }

    public DateTime? DateTime { get; set; }

    public int? NumberOfTickets { get; set; }

    public decimal? TotalPrice { get; set; }

    public int? PaymentId { get; set; }

    public string? PaymentType { get; set; }

    public virtual Payment? Payment { get; set; }

    public virtual ICollection<ReservationSeat> ReservationSeats { get; set; } = new List<ReservationSeat>();

    public virtual Screening? Screening { get; set; }

    public virtual User? User { get; set; }
}

File: CinemaNow/UI/cinemanow_mobile/lib/models/reservation.dart
import 'package:cinemanow_mobile/models/reservation_seat.dart';
import 'package:cinemanow_mobile/models/screening.dart';
import 'package:cinemanow_mobile/models/user.dart';
import 'package:json_annotation/json_annotation.dart';

part 'reservation.g.dart';

@JsonSerializable()
class Reservation {
  int? id;
  int? userId;
  int? screeningId;
  DateTime? dateTime;
  int? numberOfTickets;
  double? totalPrice;
  List<ReservationSeat>? seats;
  User? user;
  Screening? screening;

  Reservation({
    this.id,
    this.userId,
    this.screeningId,
    this.dateTime,
    this.numberOfTickets,
    this.totalPrice,
    this.seats,
    this.user,
    this.screening,
  });

  factory Reservation.fromJson(Map<String, dynamic> json) =>
      _$ReservationFromJson(json);
  Map<String, dynamic> toJson() => _$ReservationToJson(this);
}
